#348 cleanup if Spring components

marcoebbinghaus opened the pull request:
I did everything that was mentioned for that issue. The classes that still have an @Inject annotation can't be replaced trivially.

marcoebbinghaus committed:
#348 cleanup if Spring components

Brutus5000 reviewed (commented):
Can you explain why / where you think replacing the constructor with `@RequiredArgsConstructor` is not possible?
As far as I can see the classes
- ClanService
- AchievementService
- ClansController
- FeaturedModEnricher
do not do anything special in the constructor, so it should work here.

marcoebbinghaus committed:
#348 remove unused import in ClanService

marcoebbinghaus committed:
apply PR feedback

- replaced constructor from AchievementService with lombok annotation.
- replaced constructor from ClanService with lombok annotation and aligned test.

marcoebbinghaus commented:
AchievementService: I think I just missed to replace the constructor here, sorry for that.
ClanService: The ClanServiceTest made usage of the constructor. But I removed the constructor and aligned the test now so it works without the constructor.
ClansController: I did replace the constructor here, what exactly do you miss?
FeaturedModEnricher: FeaturedModEnricher does not have a constructor and it only has a static field. I think Lombok annotations do not work with static fields ("All static fields will be skipped entirely (not considered for any of the generated methods, and no setter/getter will be made for them)." from https://projectlombok.org/features/Data).

I pushed the according changes.
